src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = getIndex(key);
+        if (size >= threshold) {
+            resize();
+        }
+        putValue(index, newNode, table);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (key == null && currentNode.key == null
+                    || key != null && key.equals(currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] newTable = new Node[table.length << 1];
+        currentCapacity = newTable.length;
+        threshold = (int) (currentCapacity * DEFAULT_FACTOR);
+        size = 0;
+        table = transferData(table, newTable);
+    }
+
+    private Node<K, V>[] transferData(Node<K, V>[] oldTab, Node<K, V>[] newTab) {
+        Node<K, V> currentNode;
+        Node<K, V> transferedNode;
+        int bucketIndex;
+        for (int i = 0; i < oldTab.length; i++) {
+            currentNode = oldTab[i];
+            while (currentNode != null) {
+                transferedNode = getNewNode(currentNode);
+                bucketIndex = getIndex(transferedNode.key);
+                putValue(bucketIndex, transferedNode, newTab);
+                currentNode = currentNode.next;
+            }
+        }
+        return newTab;
+    }
+
+    private Node<K, V> getNewNode(Node<K, V> oldNode) {
+        K key = oldNode.key;
+        V value = oldNode.value;;
+        return new Node<>(key, value);
+    }
+
+    private int getIndex(Object key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % currentCapacity;
+    }
+
+    private void putValue(int bucketIndex, Node<K, V> newNode, Node<K, V>[] dstArray) {
+        if (dstArray[bucketIndex] == null) {
+            dstArray[bucketIndex] = newNode;
+            size++;
+        } else {
+            addOrChangeNode(bucketIndex, newNode, dstArray);
+        }
+    }
+
+    private void addOrChangeNode(int bucketIndex, Node<K, V> newNode, Node<K, V>[] dstArray) {
+        Node<K, V> currentNode = dstArray[bucketIndex];
+        while (currentNode != null) {
+            if (currentNode.key == null && newNode.key == null
+                    || currentNode.key != null && currentNode.key.equals(newNode.key)) {
+                currentNode.value = newNode.value;
+                break;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = newNode;
+                size++;
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+    }
+
+    static class Node<K, V> {
```

Comment:
Private methods and classes should be after public ones in your class. Consider moving the Node class to the end of the MyHashMap class.